x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/expression/function/scalar/string/StartsWithFunctionPipeTests.java

```diff
@@ -0,0 +1,127 @@
+        newB = new StartsWithFunctionPipe(
+                newLoc,
+                b2.expression(),
+                b2.field(),
+                b2.pattern(),
+                b2.isCaseSensitive());
+
+        assertEquals(newB,
+                b2.transformPropertiesOnly(v -> Objects.equals(v, b2.source()) ? newLoc : v, Source.class));
+    }
+
+    @Override
+    public void testReplaceChildren() {
+        StartsWithFunctionPipe b = randomInstance();
+        Pipe newField = pipe(((Expression) randomValueOtherThan(b.field(), () -> randomStringLiteral())));
+        Pipe newPattern = pipe(((Expression) randomValueOtherThan(b.pattern(), () -> randomStringLiteral())));
+        
+        StartsWithFunctionPipe newB = new StartsWithFunctionPipe(b.source(), b.expression(), b.field(), b.pattern(), b.isCaseSensitive());
+        StartsWithFunctionPipe transformed = (StartsWithFunctionPipe) newB.replaceChildren(newField, b.pattern());
+        assertEquals(transformed.field(), newField);
+        assertEquals(transformed.source(), b.source());
+        assertEquals(transformed.expression(), b.expression());
+        assertEquals(transformed.pattern(), b.pattern());
+        
+        transformed = (StartsWithFunctionPipe) newB.replaceChildren(b.field(), newPattern);
+        assertEquals(transformed.field(), b.field());
+        assertEquals(transformed.source(), b.source());
+        assertEquals(transformed.expression(), b.expression());
+        assertEquals(transformed.pattern(), newPattern);
+        
+        transformed = (StartsWithFunctionPipe) newB.replaceChildren(newField, newPattern);
+        assertEquals(transformed.field(), newField);
+        assertEquals(transformed.source(), b.source());
+        assertEquals(transformed.expression(), b.expression());
+        assertEquals(transformed.pattern(), newPattern);
+    }
+
+    @Override
+    protected StartsWithFunctionPipe mutate(StartsWithFunctionPipe instance) {
+        List<Function<StartsWithFunctionPipe, StartsWithFunctionPipe>> randoms = new ArrayList<>();
+        randoms.add(f -> new StartsWithFunctionPipe(f.source(),
+                f.expression(),
+                pipe(((Expression) randomValueOtherThan(f.field(), () -> randomStringLiteral()))),
+                f.pattern(),
+                f.isCaseSensitive()));
+        randoms.add(f -> new StartsWithFunctionPipe(f.source(),
+                f.expression(),
+                f.field(),
+                pipe(((Expression) randomValueOtherThan(f.pattern(), () -> randomStringLiteral()))),
+                f.isCaseSensitive()));
+        randoms.add(f -> new StartsWithFunctionPipe(f.source(),
+                f.expression(),
+                pipe(((Expression) randomValueOtherThan(f.field(), () -> randomStringLiteral()))),
+                pipe(((Expression) randomValueOtherThan(f.pattern(), () -> randomStringLiteral()))),
+                f.isCaseSensitive()));
+        
```

Comment:
Could you please add test with just changing the case sensitive flag?
and maybe one more where all of field/pattern/caseSensitive/flag are changed.